etc/kayobe/inventory/group_vars/all/firewall

```diff
@@ -0,0 +1,323 @@
+# compute_firewalld_zones: "{{ stackhpc_firewalld_zones }}" in compute.yml
+stackhpc_firewalld_zones: |
+  {% set network_zones = [] %}
+  {% for network in network_interfaces %}
+  {% if network | net_zone %}
+  {% set _ = network_zones.append({'zone': network | net_zone }) %}
+  {% endif %}
+  {% endfor %}
+  {{ network_zones | unique | list }}
+
+# A templated list of firewalld rules, according to the enabled services,
+# host's group membership, and host's network configuration.
+# This variable can be used to set any *_firewalld_rules kayobe variable e.g.
+# compute_firewalld_rules: "{{ stackhpc_firewalld_rules }}" in compute.yml
+stackhpc_firewalld_rules: |
+  {% set stackhpc_firewalld_rules_verified = stackhpc_firewalld_rules_extra %}
+  {% for rule in stackhpc_firewalld_rules_default | unique %}
+  {% if rule | ansible.utils.remove_keys('state') not in stackhpc_firewalld_rules_verified | map('ansible.utils.remove_keys', 'state') | default([]) %}
+  {% if rule.network is not defined %}
+  {% set _ = stackhpc_firewalld_rules_verified.append(rule) %}
+  {% elif rule.network in network_interfaces and rule.network | net_zone %}
+  {% set _ = stackhpc_firewalld_rules_verified.append(rule) %}
+  {% endif %}
+  {% endif %}
+  {% endfor %}
+  {{ stackhpc_firewalld_rules_verified | list }}
+
+# A single list of default firewall rules, combining the enabled rules from
+# stackhpc_firewalld_rules_template. This variable is designed to fail to
+# template if any conflicting rules are found.
+stackhpc_firewalld_rules_default: |
+  {% set stackhpc_firewalld_rules_formatted = [] %}
+  {% for ruleset in stackhpc_firewalld_rules_template %}
+  {% if ruleset.enabled | bool %}
+  {% for rule in ruleset.rules %}
+  {% if rule.zone is not defined %}
+  {% set rule = rule | combine({'zone': rule.network | net_zone }) %}
+  {% endif %}
+  {% if rule not in stackhpc_firewalld_rules_formatted %}
+  {% if rule | ansible.utils.remove_keys('state') in stackhpc_firewalld_rules_formatted | map('ansible.utils.remove_keys', 'state') %}
+  {% set _ = stackhpc_firewalld_rules_formatted.append({'state':'failure'}) %}
+  {% elif rule.network is not defined %}
+  {% set _ = stackhpc_firewalld_rules_formatted.append(rule) %}
+  {% elif rule.network in network_interfaces and rule.network | net_zone %}
+  {% set _ = stackhpc_firewalld_rules_formatted.append(rule) %}
+  {% endif %}
+  {% endif %}
+  {% endfor %}
+  {% endif %}
+  {% endfor %}
+  {{ undef(hint='ERROR: Conflicting firewall rules found') if ({'state':'failure'} in stackhpc_firewalld_rules_formatted) else stackhpc_firewalld_rules_formatted }}
+
+stackhpc_firewalld_rules_template: |
+  {{ stackhpc_common_firewalld_rules_template +
+     (stackhpc_controller_firewalld_rules_template if 'controllers' in group_names else []) +
+     (stackhpc_compute_firewalld_rules_template if 'compute' in group_names else []) +
+     (stackhpc_storage_firewalld_rules_template if 'storage' in group_names else []) +
```

Comment:
Consider changing to this to avoid issues when Ceph runs on controllers or computes:

```
 (stackhpc_ceph_firewalld_rules_template if 'ceph' in group_names else []) +
```